ISISScientificComputing/autoreduce
# New End of Run Monitor

### Summary of work

Implementation of the new end of run monitor design.

### How to test your work

See updated README file for details on how to setup locally.

Unit tests can be run without any setup. Just type ```pytest test_new_end_of_run_monitor.py```

### Additional comments

I have made the cycle folder a setting that has to be manually adjusted. The reason I did this is to minimise the knowledge of the archive that's written into the code itself, this helps with testing because we don't have to go about setting up fake archives. At the same time, I thought it would be too cumbersome to have to adjust the cycle folder on every instrument in the CSV file, so I put it in settings.py. Let me know what you think is best.

I experienced difficulty with the EoRM logging config being overwritten by that of the queue client. That's why I have set up a logger in a slightly different way.

Fixes #299
